webapp/src/main/java/com/box/l10n/mojito/service/scheduledjob/jobs/ScheduledThirdPartySync.java

```diff
@@ -0,0 +1,146 @@
+package com.box.l10n.mojito.service.scheduledjob.jobs;
+
+import com.box.l10n.mojito.entity.ScheduledJob;
+import com.box.l10n.mojito.pagerduty.PagerDutyException;
+import com.box.l10n.mojito.pagerduty.PagerDutyIntegrationService;
+import com.box.l10n.mojito.pagerduty.PagerDutyPayload;
+import com.box.l10n.mojito.quartz.QuartzPollableTaskScheduler;
+import com.box.l10n.mojito.service.pollableTask.PollableFuture;
+import com.box.l10n.mojito.service.scheduledjob.IScheduledJob;
+import com.box.l10n.mojito.service.scheduledjob.ScheduledJobRepository;
+import com.box.l10n.mojito.service.thirdparty.ThirdPartySyncJob;
+import com.box.l10n.mojito.service.thirdparty.ThirdPartySyncJobInput;
+import com.box.l10n.mojito.utils.ServerConfig;
+import com.google.common.collect.ImmutableMap;
+import org.apache.commons.lang.text.StrSubstitutor;
+import org.quartz.DisallowConcurrentExecution;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.util.UriComponentsBuilder;
+
+/**
+ * Scheduled job that performs the third party sync for a repository off a cron schedule.
+ *
+ * @author mattwilshire
+ */
+@Component
+@DisallowConcurrentExecution
+public class ScheduledThirdPartySync implements IScheduledJob {
+
+  static Logger logger = LoggerFactory.getLogger(ScheduledThirdPartySync.class);
+
+  @Autowired private QuartzPollableTaskScheduler quartzPollableTaskScheduler;
+  @Autowired private ScheduledJobRepository scheduledJobRepository;
+  @Autowired private PagerDutyIntegrationService pagerDutyIntegrationService;
+
+  @Autowired ServerConfig serverConfig;
+
+  @Value(
+      "${l10n.scheduledJobs.thirdPartySync.notifications.title:MOJITO | Third party sync failed for {repository}}")
+  String notificationTitle;
+
+  private ScheduledJob scheduledJob;
```

Comment:
Also a possible misunderstanding but why do we need to store these variables as class level variables? Does the context not have the details to find them again? 🤔  

 I don't know this library but I'm mainly worried about side-effects.

Reply:
The notification title is pulled in from the application properties, you can have different PagerDuty titles for different environments. The ScheduledJob here is set on execution, we pull it out of the database on the execute method, when the success or failure method is called by the listener we can reference the job as its tied to the instance.